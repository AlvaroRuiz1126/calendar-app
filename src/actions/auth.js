import Swal from "sweetalert2";
import { fetchWhithoutToken, fetchWhithToken } from "../helpers/fetch";
import { types } from "../types/types";

export const startLogin = (email, password) => {
    return async(dispatch) => {
        //console.log(email, password);
        const resp = await fetchWhithoutToken('auth', {email, password}, 'POST');
        const body = await resp.json();
        //console.log(body);

        if(body.ok){
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        }else{
            Swal.fire('Error', body.msg, 'error');
        }
    };
};

export const startRegister = (email, password, name) => {
    return async(dispatch) => {
        const resp = await fetchWhithoutToken('auth/new', {email, password, name}, 'POST');
        const body = await resp.json();
        //console.log(body);

        if(body.ok){
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        }else{
            Swal.fire('Error', body.msg, 'error');
        }
    };
};

export const startChecking = () => {
    return async(dispatch) => {
        const resp = await fetchWhithToken('auth/renew');
        const body = await resp.json();
        //console.log(body);

        if(body.ok){
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        }else{
            //Swal.fire('Error', body.msg, 'error');
            dispatch(checkCheckingFinish());
        }

    };
};

export const startLogout = () => {
    return (dispatch) => {
        localStorage.clear();
        dispatch(logout());
    };
}

const login = (user) => ({
    type: types.authLogin,
    payload: user
});

const checkCheckingFinish = () => ({
    type: types.authChekingFinish
});

const logout = () => ({type: types.authLogout});