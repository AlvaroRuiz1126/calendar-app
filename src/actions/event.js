import Swal from "sweetalert2";
import { fetchWhithToken } from "../helpers/fetch";
import { prepareEvents } from "../helpers/prepareEvents";
import { types } from "../types/types";

export const eventStartAddNew = (event) => {
    return async(dispatch, getState) => {
        //console.log(event);
        const {uid, name} = getState().auth;
        try {
            const resp = await fetchWhithToken('events', event, 'POST');
            const body = await resp.json();
            console.log(body);
            
            if(body.ok){
                event.id = body.event.id
                event.user = {
                    _id: uid,
                    name
                };

                console.log(event);
                dispatch(eventAddNew(event));
                dispatch(eventStartLoading());
            }

        } catch (error) {
            console.log(error);
        }
    };
};

const eventAddNew = (event) => ({
    type: types.eventAddNew,
    payload: event
});

export const eventSetActive = (event) => ({
    type: types.eventSetActive,
    payload: event
});

export const eventClearActiveEvent = () => ({
    type: types.eventClearActiveEvent
});

export const eventStartUpdated = (event) => {
    return async(dispatch) => {
        //console.log(event);
        try {
            const resp = await fetchWhithToken(`events/${event.id}`, event, 'PUT');
            const body = await resp.json();

            if(body.ok){
                dispatch(updated(event));
            }else{
                Swal.fire('Error', body.msg, 'error');
            }
            
        } catch (error) {
            console.log(error);
        }
    };
};

const updated = (event) => ({
    type: types.eventUpdated,
    payload: event
});

export const eventStartDelete = (event) => {
    return async(dispatch, getState) => {
        const {id} = getState().calendar.activeEvent
        try {
            const resp = await fetchWhithToken(`events/${id}`, {}, 'DELETE');
            const body = await resp.json();
            
            if(body.ok){
                dispatch(eventDeleted());
            }else{
                Swal.fire('Error', body.msg, 'error');
            }

        } catch (error) {
            console.log(error);
        }


    };
};

const eventDeleted = () => ({
    type: types.eventDeleted
});

export const eventStartLoading = () => {
    return async(dispatch) => {
        const resp = await fetchWhithToken('events');
        const body = await resp.json();
        //console.log(body);

        const events = prepareEvents(body.events);
        console.log(events);

        //dispatch(eventLoaded(body.events));
        dispatch(eventLoaded(events));
    };
};

const eventLoaded = (events) => ({
    type: types.eventLoaded,
    payload: events
});

export const clearEvents = () => ({type: types.eventLogout});