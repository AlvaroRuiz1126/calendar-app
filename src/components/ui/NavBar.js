import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startLogout } from '../../actions/auth';
import { clearEvents } from '../../actions/event';

export const NavBar = () => {
    const dispatch = useDispatch();
    const {name} = useSelector(state => state.auth);

    const handleLogout = () => {
        console.log('Salir');
        dispatch(startLogout());
        dispatch(clearEvents());
    };

    return (
        <div className="navbar navbar-dark bg-dark mb-5">
            <span className="navbar-brand">
                {name}
            </span>

            <button 
                className="btn btn-outline-danger"
                onClick={handleLogout}
            >
                <i className="fas fa-sign-out-alt"></i>
                <span> Salir</span>
            </button>
            
        </div>
    )
}
