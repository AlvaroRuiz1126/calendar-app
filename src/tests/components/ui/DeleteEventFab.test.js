import '@testing-library/jest-dom';
import React from 'react';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux';
import { DeleteEventFab } from '../../../components/ui/DeleteEventFab';
import { mount } from 'enzyme';
import { eventStartDelete } from '../../../actions/event';
jest.mock('../../../actions/event', () => ({
    eventStartDelete: jest.fn()
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {};
const store = mockStore(initState);
store.dispatch = jest.fn();
const wrapper = mount(
    <Provider store={store}>
        <DeleteEventFab />
    </Provider>
);

describe('Pruebas en el componente DeleteEventFab', () => {
    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe llamar el event startDelete cuando se de click en el botÃ³n', () => {
        wrapper.find('button').prop('onClick')();
        
        expect(store.dispatch).toHaveBeenCalledTimes(1);
        expect(eventStartDelete).toHaveBeenCalled();
    });
});