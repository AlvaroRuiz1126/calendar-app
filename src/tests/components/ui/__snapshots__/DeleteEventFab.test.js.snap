// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente DeleteEventFab Debe mostrar el componente correctamente 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Borrar Evento
  </span>
</button>
`;
